pkg/kosmosctl/unjoin/unjoin.go

```diff
@@ -0,0 +1,169 @@
+
+	cmd := &cobra.Command{
+		Use:                   "unjoin",
+		Short:                 i18n.T("Unjoin this in clusterlink control plane"),
+		Long:                  "",
+		Example:               unjoinExample,
+		SilenceUsage:          true,
+		DisableFlagsInUseLine: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctlutil.CheckErr(o.Complete(f, cmd, args))
+			ctlutil.CheckErr(o.Validate())
+			ctlutil.CheckErr(o.Run(f, cmd, args))
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&o.MasterKubeConfig, "master-kubeconfig", "", "", "master-kubeconfig")
+	cmd.Flags().StringVarP(&o.ClusterKubeConfig, "cluster-kubeconfig", "", "", "cluster-kubeconfig")
+	err := cmd.MarkFlagRequired("cluster-kubeconfig")
+	if err != nil {
+		fmt.Printf("kosmosctl join cmd error, MarkFlagRequired failed: %s", err)
+	}
+	return cmd
+}
+func (o *CommandUnJoinOptions) Complete(f ctlutil.Factory, cmd *cobra.Command, args []string) error {
+	var masterConfig *restclient.Config
+	var err error
+
+	if o.MasterKubeConfig != "" {
+		masterConfig, err = clientcmd.BuildConfigFromFlags("", o.MasterKubeConfig)
+		if err != nil {
+			return fmt.Errorf("kosmosctl unjoin complete error, generate masterConfig failed: %s", err)
+		}
+	} else {
+		masterConfig, err = f.ToRESTConfig()
+		if err != nil {
+			return fmt.Errorf("kosmosctl unjoin complete error, get current masterConfig failed: %s", err)
+		}
+	}
+
+	clusterConfig, err := clientcmd.BuildConfigFromFlags("", o.ClusterKubeConfig)
+	if err != nil {
+		return fmt.Errorf("kosmosctl unjoin complete error, generate memberConfig failed: %s", err)
+	}
+
+	o.Client, err = kubernetes.NewForConfig(clusterConfig)
+	if err != nil {
+		return fmt.Errorf("kosmosctl join complete error, generate basic client failed: %v", err)
+	}
+
+	o.DynamicClient, err = dynamic.NewForConfig(masterConfig)
+	if err != nil {
+		return fmt.Errorf("kosmosctl unjoin complete error, generate dynamic client failed: %s", err)
+	}
+
+	return nil
+}
+
+func (o *CommandUnJoinOptions) Validate() error {
+	return nil
+}
+
+func (o *CommandUnJoinOptions) Run(f ctlutil.Factory, cmd *cobra.Command, args []string) error {
+	//delete cluster
+	err := o.DynamicClient.Resource(schema.GroupVersionResource{
+		Group:    "kosmos.io",
+		Version:  "v1alpha1",
+		Resource: "clusters",
+	}).Namespace("").Delete(context.TODO(), args[0], metav1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("(cluster) kosmosctl unjoin run error, delete cluster failed: %s", err)
+	}
+
+	for {
+		// Wait 3 second
+		time.Sleep(3 * time.Second)
+		_, err := o.DynamicClient.Resource(schema.GroupVersionResource{
+			Group:    "kosmos.io",
+			Version:  "v1alpha1",
+			Resource: "clusters",
+		}).Namespace("").Get(context.TODO(), args[0], metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				break
+			}
+			return fmt.Errorf("(cluster) kosmosctl unjoin run error, delete cluster failed: %s", err)
+		}
```

Comment:
put time.Sleep(3 * time.Second)  here may be better

Reply:
why not first wait 3 second. When I test delete cluster, I usually can't delete it right away.need wait time

Reply:
you can refer https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/
some controller will add finalizer on cluster, let cluster can not delete immediately

Reply:
because cluster have finalizer, i think that we should wait before the first get.